File: flapypan-blog-api/src/main/java/top/flapypan/blog/common/RestResult.java
package top.flapypan.blog.common;

import com.fasterxml.jackson.annotation.JsonIgnore;
import lombok.AllArgsConstructor;
import lombok.Getter;
import org.springframework.http.HttpStatus;

/**
 * 统一返回结构
 */
@AllArgsConstructor
public class RestResult<T> {
    @JsonIgnore
    public static final int OK_STATUS = HttpStatus.OK.value();

    @Getter
    private int code;
    @Getter
    private T data;

    public boolean isSuccess() {
        return code == OK_STATUS;
    }

    public static <T> RestResult<T> success() {
        return new RestResult<>(OK_STATUS, null);
    }

    public static <T> RestResult<T> success(T data) {
        return new RestResult<>(OK_STATUS, data);
    }

    public static <T> RestResult<T> failure(int code) {
        return new RestResult<>(code, null);
    }

    public static <T> RestResult<T> failure(int code, T data) {
        return new RestResult<>(code, data);
    }
}

File: flapypan-blog-api/src/main/kotlin/top/flapypan/blog/common/RestResult.kt
package top.flapypan.blog.common

import org.springframework.http.HttpStatus

private val OK_STATUS = HttpStatus.OK.value()

/**
 * 统一返回结构
 */
data class RestResult<T>(
    val code: Int,
    val data: T,
    val success: Boolean = (code == OK_STATUS)
)

/**
 * 扩展方法，用于包装返回类型
 */
fun <T> T?.restOk() = RestResult(OK_STATUS, this)

/**
 * 扩展方法，用于包装返回类型
 */
fun <T> T?.restErr(code: Int = HttpStatus.INTERNAL_SERVER_ERROR.value()) = RestResult(code, this)
